be/src/vec/functions/function_string.h

```diff
@@ -1344,6 +1345,167 @@ class FunctionSplitPart : public IFunction {
+        ColumnArray::Offset64 src_offsets_size = src_column_string->get_offsets().size();
+        NullMapType null_map(src_null_map->size());
+        for (size_t i = 0; i < src_null_map->size(); ++i) {
+            null_map[i] = (*src_null_map)[i];
+        }
+
+        for (size_t i = 0; i < src_offsets_size; i++) {
+            if (src_null_map && null_map[i]) {
+                (*dest_nested_null_map).push_back(true);
+                column_string_offsets.push_back(string_pos);
+                dest_offsets.push_back(dest_pos);
+                continue;
+            }
+
+            const auto delimiter = delimiter_column.get_data_at(i).to_string();
+            const auto str = src_column_string->get_data_at(i).to_string();
+            StringRef str_ref = src_column_string->get_data_at(i);
+
+            if (str.size() == 0) {
+                dest_offsets.push_back(dest_pos);
+                if (src_null_map) {
+                    (*dest_nested_null_map).push_back(false);
+                }
+                continue;
+            }
+
+            if (delimiter.size() == 0) {
+                //If there is no delimiter, the entire string is printed
+                const size_t old_size = column_string_chars.size();
+                const size_t new_size = old_size + str.size();
+                const size_t str_size = str.size();
+                column_string_chars.resize(new_size);
+                if (str_size > 0) {
+                    memcpy(column_string_chars.data() + old_size, str_ref.data, str_ref.size);
+                }
+                if (src_null_map) {
+                    (*dest_nested_null_map).push_back(false);
+                }
+                string_pos += str_size;
+                dest_pos++;
+                column_string_offsets.push_back(string_pos);
+            } else if (delimiter.size() == 1) {
+                for (size_t str_pos = 0; str_pos <= str.size();) {
+                    const size_t str_offset = str_pos;
+                    const size_t old_size = column_string_chars.size();
+                    const size_t split_part_size = split_str(str_pos, str, delimiter);
+                    str_pos++;
+                    const size_t new_size = old_size + split_part_size;
+                    column_string_chars.resize(new_size);
+                    if (split_part_size > 0) {
+                        memcpy(column_string_chars.data() + old_size, str_ref.data + str_offset,
+                               split_part_size);
+                    }
+                    if (src_null_map) {
+                        (*dest_nested_null_map).push_back(false);
+                    }
+                    string_pos += split_part_size;
+                    dest_pos++;
+                    column_string_offsets.push_back(string_pos);
+                }
+            }
+            dest_offsets.push_back(dest_pos);
+        }
+    }
+
+    size_t split_str(size_t& pos, const string str, string delimiter) {
```

Comment:
the delimiter should be `char`.
the str should be `string_ref` or `string_value`

Reply:
I have fixed other codes but I don't understand here why use `string_ref` or `string_value` instead of `char`. Can you explain it? Appreciate that!